src/Count.php

```diff
@@ -22,38 +22,50 @@ final class Count
     private int $count;
 
     public function __construct(
-        float $count
+        int $count
     ) {
+        $this->count = $count;
+
         Assert::lessThan(
-          $count,
-          self::MAX_COUNT,
-          'The maximum number of views has been reached'
+            $count,
+            self::MAX_COUNT,
+            'The maximum number of views has been reached',
         );
-        $this->count = intval($count);
         Assert::greaterThanEq(
-          $count,
-          0,
-          "Received a negative number of views"
+            $count,
+            0,
+            'Number of views cannot be negative',
         );
     }
 
-    public static function ofString(string $countStr): self
-    {
-      Assert::digits(
-        $countStr,
-        'The base count must only contain digits'
-      );
-      $count = floatval($countStr);
-      return new self($count);
+    public static function ofString(
+        string $value
+    ): self {
+        Assert::digits(
+            $value,
+            'The base count must only contain digits',
+        );
+        $count = intval($value);
+
+        return new self($count);
     }
 
     public function toInt(): int
     {
         return $this->count;
     }
 
-    public function plus(self $count): self
-    {
-      return new self($this->toInt() + $count->toInt());
+    public function plus(
+        self $that
+    ): self {
+        $sum = $this->toInt() + $that->toInt();
+
+        if (!is_int($sum)) {
+            throw new \InvalidArgumentException(
+                'The maximum number of views has been reached',
```

Comment:
With this check, we currently have two sources of truth for checking whether the maximum number of views has been reached: the `is_int` check here and the `assert less than MAX_COUNT` assertion in the constructor. We know this wouldn't cause logical problems since `MAX_COUNT` will always be <= `PHP_INT_MAX`, but maybe it would be better if they become a single source of truth instead?

If we are sure that `MAX_COUNT` is always going to be `PHP_INT_MAX` and won't change in the future, then maybe we can remove the assertion in the constructor (along with `MAX_COUNT`) since it only receives ints and compares them against `PHP_INT_MAX` (ints will never exceed `PHP_INT_MAX`).

If `MAX_COUNT` might change in the future to be less than `PHP_INT_MAX`, then maybe we should find a way to do the check in one place or to do the same check here and in the constructor.

With the implementation in the PR, if `MAX_COUNT` changes, the message "the maximum number of views has been reached" in the `plus` function implies a different meaning than the same message in the constructor (albeit that doesn't cause any logical problems).

Reply:
> If `MAX_COUNT` might change in the future to be less than `PHP_INT_MAX`, then maybe we should find a way to do the check in one place or to do the same check here and in the constructor.

Or maybe, in that case, we can change the error message in the `plus` function to something along the lines of: "the number of views have exceeded the integer limit". This way we wouldn't have one error message having two different meanings (exceeding the integer limit and exceeding `MAX_COUNT`).

Reply:
Yeah this kinda bad design for now, but in reality I'm not sure we will ever reach the max value 😅

Reply:
It's just a protection vs type error in cases when people will try to send too big base value.

Reply:
Right now there are a lot of fraud requests from users, trying to avoid error log from the spam 😞

Reply:
I'd suggest to push it as is and rethink it later.

Reply:
Cool